src/Propel/Generator/Reverse/PgsqlSchemaParser.php

```diff
@@ -417,7 +417,10 @@ protected function addForeignKeys(Table $table, $oid)
                 $fk = new ForeignKey($name);
                 $fk->setForeignTableCommonName($foreignTable->getCommonName());
                 if ($table->guessSchemaName() != $foreignTable->guessSchemaName()) {
-                    $fk->setForeignSchemaName($foreignTable->getSchema());
+                     if($foreignTable->getSchema())
+                            $fk->setForeignSchemaName($foreignTable->getSchema());
+                     elseif($foreignTable->getSchema() == 'public')
```

Comment:
Why shouldn't it work? If the condition is `false` then we have no `schema` for the foreignTable, that's it (although there should always be a schema).
But if it evaluates to `true` AND it is equal to public, then the `foreignSchemaName` will be set to public. (Maybe instead of `elseif` I should have used only `if`).
Without it, the `public` schema will not be set.